### PR TITLE
MODFQMMGR-309 Add simple_instance_type

### PR DESCRIPTION
**Pull request by mweaver-ebsco**
## Pre-Merge Checklist

If you are adding entity type(s), have you:
- [X] Added the JSON5 definition to the `src/main/resources/entity-types` directory?
- [X] Ensured that GETing the entity type at `/entity-types/{id}` works as expected?
- [X] Added translations for all fields, per the [translation guidelines](/translations/README.md)? (Check this by ensuring `GET /entity-types/{id}` does not have `mod-fqm-manager.entityType.` in the response)
- [X] Added views to liquibase, as applicable?
- [X] Added required interfaces to the module descriptor?
- [X] Checked that querying fields works correctly and all SQL is valid?

### CODE REVIEW

File: src/main/resources/entity-types/inventory/composite_instances.json5 [hunk 1]
```diff
@@ -16,7 +16,7 @@
       join: {
         type: 'left join',
         joinTo: 'instance',
-        condition: ":this.id = (:that.jsonb ->> 'statusId')::uuid"
+        condition: ":this.id = :that.instanceStatusId"
```
Comment (mweaver-ebsco) [inline src/main/resources/entity-types/inventory/composite_instances.json5:19]: These join condition changes are just something I noticed when adding the new source. Apparently the `instance` table has indexes on all of its non-jsonb columns and not on the corresponding jsonb fields, so we should see a pretty big performance boost from this change

File: src/main/resources/entity-types/inventory/simple_instance_type.json5 [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+      valueFunction: 'lower(${tenant_id}_mod_inventory_storage.f_unaccent (:value))',
+      source: {
+        entityTypeId: 'af44e2e0-12e0-4eec-b80d-49feb33a866c',
+        columnName: 'code',
+      },
+      valueSourceApi: {
+        path: 'instance-types',
+        valueJsonPath: '$.instanceTypes.*.code',
+        labelJsonPath: '$.instanceTypes.*.code',
+      }
```
Comment (mweaver-ebsco) [inline src/main/resources/entity-types/inventory/simple_instance_type.json5:59]: These fields with `valueSourceApi`s all have backing indexes, so there's no need to redirect them to the `id` field with `idColumnName`

File: src/main/resources/entity-types/inventory/simple_instance_type.json5 [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+      target: 'src_inventory_instance_type',
+    },
+  ],
+  requiredPermissions: [
+    'inventory-storage.instance-types.collection.get',
+    'inventory-storage.instance-types.item.get'
+  ],
+  columns: [
+    {
+      name: 'id',
```
Comment (mweaver-ebsco) [inline src/main/resources/entity-types/inventory/simple_instance_type.json5:18]: I opted not to bother adding a `valueSourceApi` for `id` even though we could. I figure that users wouldn't really benefit from a list of UUIDs (especially when they can use `code` or `name` to search easily). I assume that if someone were to search by ID, they would probably be copying/pasting from somewhere. And in that case, there's a solid chance that they are searching for an ID that doesn't exist.

File: translations/mod-fqm-manager/en.json [hunk 4]
```diff
@@ -1141,6 +1142,15 @@
   "entityType.simple_instance.updated_at": "Updated at",
   "entityType.simple_instance.updated_by": "Updated by user UUID",
   "entityType.simple_instance.version": "Record version",
+  "entityType.simple_instance_type": "Instance type",
+  "entityType.simple_instance_type.metadata_updated_by_user_id": "Updated by user UUID",
+  "entityType.simple_instance_type.metadata_created_by_user_id": "Created by user UUID",
```
Comment (mweaver-ebsco) [inline translations/mod-fqm-manager/en.json:1147]: Stupid :(

File: translations/mod-fqm-manager/en.json [hunk 5]
```diff
@@ -1141,6 +1142,15 @@
...
+  "entityType.simple_instance_type.metadata_updated_date": "Updated date",
+  "entityType.simple_instance_type.metadata_created_date": "Crated date",
+  "entityType.simple_instance_type.source": "Source",
+  "entityType.simple_instance_type.name": "Name",
+  "entityType.simple_instance_type.code": "Code",
+  "entityType.simple_instance_type.id": "Instance UUID",
```
Comment (mweaver-ebsco) [inline translations/mod-fqm-manager/en.json:1153]: Really stupid :(

File: translations/mod-fqm-manager/en.json [hunk 6]
```diff
@@ -231,6 +231,7 @@
   "entityType.composite_instances.instance": "Instance",
   "entityType.composite_instances.inst_stat": "Instance status",
   "entityType.composite_instances.mode_of_issuance": "Mode of issuance",
+  "entityType.composite_instances.instance_type": "Resource type",
```
Comment (mweaver-ebsco) [inline translations/mod-fqm-manager/en.json:234]: Technically, this doesn't exactly follow the requirements laid out in MODFQMMGR-309 (which is just asking for the instance type name, and calling _that_ "Resource type"), but I figured that it's probably better to give that name to the whole child ET

File: translations/mod-fqm-manager/en.json (hunk 5 above)
Comment (kjain110) [reply translations/mod-fqm-manager/en.json:1153]: :(

File: src/main/resources/entity-types/inventory/simple_instance_type.json5 [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+      sourceAlias: 'instance_type',
+      dataType: {
+        dataType: 'rangedUUIDType',
+      },
+      isIdColumn: true,
+      queryable: true,
+      visibleByDefault: false,
+      valueGetter: ":sourceAlias.jsonb->>'id'",
```
Comment (bvsharp) [inline src/main/resources/entity-types/inventory/simple_instance_type.json5:26]: would we be able to just use :sourceAlias.id here?

**Review by bvsharp — APPROVED** [review]

**Review by ncovercash — APPROVED** [review]

**Review by kjain110 — APPROVED** [review]

File: src/main/resources/entity-types/inventory/simple_instance_type.json5 (hunk 7 above)
Comment (mweaver-ebsco) [reply src/main/resources/entity-types/inventory/simple_instance_type.json5:26]: Nice catch! That would make it use an index, too!

File: translations/mod-fqm-manager/en.json [hunk 8]
```diff
@@ -1141,6 +1142,15 @@
   "entityType.simple_instance.updated_at": "Updated at",
   "entityType.simple_instance.updated_by": "Updated by user UUID",
   "entityType.simple_instance.version": "Record version",
+  "entityType.simple_instance_type": "Instance type",
+  "entityType.simple_instance_type.metadata_updated_by_user_id": "Updated by user UUID",
+  "entityType.simple_instance_type.metadata_created_by_user_id": "Created by user UUID",
+  "entityType.simple_instance_type.metadata_updated_date": "Updated date",
+  "entityType.simple_instance_type.metadata_created_date": "Crated date",
```
Comment (mweaver-ebsco) [inline translations/mod-fqm-manager/en.json:1149]:
```suggestion
  "entityType.simple_instance_type.metadata_created_date": "Created date",
```

**Pull request merged**